src/Controller.ts

```diff
@@ -6,14 +6,15 @@ import {Config} from './lib/util/Config';
 import {EnvOverridable, Services} from './Constants';
 import {RuleManager} from './lib/RuleManager';
 import {RuleEngine} from './lib/services/RuleEngine'
+import {DependencyChecker} from './lib/services/DependencyChecker';
 import {RulePathManager} from './lib/RulePathManager';
 
 /**
  * Converts an array of RuleEngines to a sorted, comma delimited
  * string of their names.
  */
-function enginesToString(engines: RuleEngine[]): string {
-	return engines.map(e => e.getName()).sort().join(', ');
+function servicesToString(services: (RuleEngine|DependencyChecker)[]): string {
```

Comment:
Adding a new interface `Service` with method `getName` and changing RuleEngine and DependencyChecker to extend these would be cleaner.

Reply:
Good point.